var/spack/repos/builtin/packages/openscenegraph/package.py

```diff
@@ -59,22 +97,35 @@ def cmake_args(self):
             "-DDYNAMIC_OPENSCENEGRAPH={0}".format(shared_status),
             "-DDYNAMIC_OPENTHREADS={0}".format(shared_status),
             "-DOPENGL_PROFILE={0}".format(opengl_profile),
+            self.define_from_variant("BUILD_OSG_APPLICATIONS", "apps"),
             # General Options #
-            "-DBUILD_OSG_APPLICATIONS=OFF",
             "-DOSG_NOTIFY_DISABLED=ON",
             "-DLIB_POSTFIX=",
             "-DCMAKE_RELWITHDEBINFO_POSTFIX=",
             "-DCMAKE_MINSIZEREL_POSTFIX=",
         ]
 
-        if spec.satisfies("~ffmpeg"):
-            for ffmpeg_lib in ["libavcodec", "libavformat", "libavutil"]:
-                args.extend(
-                    [
-                        "-DFFMPEG_{0}_INCLUDE_DIRS=".format(ffmpeg_lib.upper()),
-                        "-DFFMPEG_{0}_LIBRARIES=".format(ffmpeg_lib.upper()),
-                    ]
-                )
+        # explicitly disable or enable plugins depending on variants
+        # CMake will still search for the packages, but won't build the plugins requiring them
+        def build_plugin(plugin, variant=False):
+            if not variant:
+                variant = plugin
+            if spec.satisfies("+{}".format(variant)):
+                value = 1
+            else:
+                value = 0
+            return self.define("BUILD_OSG_PLUGIN_{}".format(plugin.upper()), value)
+
+        args.append(build_plugin("dicom", "dcmtk"))
+        args.append(build_plugin("exr", "openexr"))
+        args.append(build_plugin("ffmpeg"))
+        args.append(build_plugin("gdal"))
+        args.append(build_plugin("ogr", "gdal"))
+        args.append(build_plugin("gta"))
+        args.append(build_plugin("inventor"))
+        args.append(build_plugin("opencascade"))
+        args.append(build_plugin("pdf"))
+        args.append(build_plugin("svg"))
```

Comment:
Apologies for getting back to this kind of late. Can you use `self.define_from_variants` instead of redefining a similar function in the package?

Reply:
Thanks for this suggestion: while looking at the resulting diff, I saw that it totally makes sense not to reinvent this wheel!